practice_app/api/views.py

```diff
@@ -1,3 +1,49 @@
 from django.shortcuts import render
-
+import requests
 # Create your views here.
+
+from django.http import JsonResponse, HttpResponse
+
+def eric_papers(request):
+    
+    #params --> title, rows
+    default_rows = '5'
+
+    search_title = request.GET.get('title')
+    rows = request.GET.get('rows', default_rows)
+
+    if search_title == None or search_title.strip() == '':
+        return JsonResponse({'message':'A paper title must be given.'}, status=404)
+   
+    try:
+        int(rows)
+    except ValueError:
+        return JsonResponse({'message':'Row count must be valid.'}, status=404)
+
+    #response --> source, authors, id, date, abstract, title, url, position
+
+    baseURL = "https://api.ies.ed.gov/eric/"
+    response_fields = "&fields=id AND title AND author AND publicationdateyear AND url AND description AND source"
+    endpoint = baseURL + "?search=title:" + search_title + "&rows=" + rows + response_fields
+
+    response = requests.get(endpoint)
+
+    if response.status_code == 200:
+        papers = response.json()['response']['docs']
+        
+        i = 0
+        for paper in papers:
```

Comment:
this logic seems like unnnecessary in this code

Reply:
I think it is necessary for the position key.

Reply:
Sorry. I thought that it doesn't affect papers variable but it does.